src/include/units/concepts.h

```diff
@@ -122,7 +113,7 @@ inline constexpr bool is_derived_from_base_dimension = decltype(is_derived_from_
  * Satisfied by all dimension types derived from an specialization of `base_dimension`.
  */
 template<typename T>
-concept BaseDimension = detail::is_derived_from_base_dimension<T>;
+concept BaseDimension = requires { detail::to_base_base_dimension(std::declval<const volatile T*>()); };
```

Comment:
The `to_base_`* function templates could be replaced with a lambda. It doesn't work on an older than supported version of MSVC (which is the newest Compiler Explorer has): https://godbolt.org/z/Y8M3WP.